functions/Backup-DbaDatabase.ps1

```diff
@@ -71,27 +75,47 @@ Backs up AdventureWorks2014 to sql2016's C:\temp folder
 #>
 	[CmdletBinding(DefaultParameterSetName = "Default")]
 	param (
-		[parameter(ParameterSetName = "NoPipe", Mandatory = $true)]
+		[parameter(ParameterSetName = "NoPipe", Mandatory = $true)][parameter(ParameterSetName = "Pipe")]
```

Comment:
I'd pick full SMO objects over text if we had to do just one but if we'd support both, you can do a check on the object type and if it's string, ensure that -SqlInstance is specified.